module/surface_sam_sid_vhf.c

```diff
@@ -0,0 +1,411 @@
+/*
+ * Virtual HID Framwork (VHF) driver for input events via SAM.
+ * Used for keyboard input events on the Surface Laptops.
+ */
+
+#include <linux/acpi.h>
+#include <linux/hid.h>
+#include <linux/input.h>
+#include <linux/platform_device.h>
+#include <linux/types.h>
+
+#include "surface_sam_ssh.h"
+
+#define SID_VHF_INPUT_NAME		"Microsoft Virtual HID Framework Device"
```

Comment:
Based on your look at the Windows device manager I'm not sure anymore that this is the right name for the devices. Probably something like `Microsoft Surface Keyboard/Touchpad` would be better?

Reply:
Well the HID devices themselves were named really badly so I don't think we should stick to the Windows naming conventions at all:

- HID-compliant touchpad (09AF)
- HID-compliant consumer control device (09AE)
- HID-compliant vendor defined device (0922)

I think we can keep the current name, because once `input` detects the HID device, it'll create a touchpad, a keyboard and a consumer control by suffixing the names:

```
[  977.195376] input: Microsoft Virtual HID Framework Device Keyboard as /devices/platform/MSHW0114:00/surface_sam_sid_vhf/0006:045E:F001.0006/input/input35
[  977.195412] hid-multitouch 0006:045E:F001.0006: sid_vhf_hid_open
[  977.195729] input: Microsoft Virtual HID Framework Device Consumer Control as /devices/platform/MSHW0114:00/surface_sam_sid_vhf/0006:045E:F001.0006/input/input38
[  977.195839] input: Microsoft Virtual HID Framework Device Touchpad as /devices/platform/MSHW0114:00/surface_sam_sid_vhf/0006:045E:F001.0006/input/input40
[  977.196109] hid-multitouch 0006:045E:F001.0006: input,hidraw3: <UNKNOWN> HID v0.00 Keyboard [Microsoft Virtual HID Framework Device] on 
```

Reply:
Yeah, I'd drop the VHF part though since you mentioned they were different devices. Maybe something like `Microsoft Surface HID` plus the suffix? I think that will also be a bit more accessible for users as those names are visible in userspace.

Reply:
Done